### PR TITLE
Bugfix/electron logging support

### PR DESCRIPTION
**Pull request by timtmok**
### Intent
Address #10929 and #11565

### Approach
Reads from `logging.conf` and environment variables. There's an object now to represent the options and maps the command-line/logging.conf option to the environment variable.

Now that `logger-type` can be set to `stderr`, the log to console has been removed from dev builds.

Syslog has been implemented for Linux only. OSX has a syslog but it doesn't appear well documented and `winston-syslog` doesn't seem able to log to it. It might be `local0` isn't supported and `udp` must be used but changing it doesn't appear to be allowed without disabling some security permissions.

### Automated Tests
There are some unit tests to ensure the log options are set properly. It doesn't check that it can log with those options though.

### QA Notes
Linux should log to `syslog` without any configuration. It's set to use local sockets so there shouldn't be any requirement to configure a firewall.

`stderr` will be visible when running RStudio from a command-line.

`file` is the default logger type if a type is unspecified. It still follows a daily rotated log file with the name including the date.

### Checklist

- [x] If this PR adds a new feature, or fixes a bug in a previously released version, it includes an entry in `NEWS.md` 
- [x] If this PR adds or changes UI, the updated UI meets [accessibility standards](https://github.com/rstudio/rstudio/wiki/Accessibility)
- [x] A reviewer is assigned to this PR (if unsure who to assign, check Area Owners list)
- [x] This PR passes all local unit tests

### CODE REVIEW

**Review by kevinushey — APPROVED** [review]
Overall LGTM!

File: src/node/desktop/src/core/winston-logger.ts [hunk 1]
```diff
@@ -14,43 +14,58 @@
...
-          dirname: logFile.getParent().getAbsolutePath(),
-          datePattern: 'YYYY-MM-DD',
-          frequency: '1d',
-        })
-      );
+    let logTransport;
+    if (logOptions.getLoggerType() === 'stderr') {
+      logTransport = new Console();
+    } else if (logOptions.getLoggerType() === 'syslog') {
+      if (process.platform === 'linux') {
```
[the remark this answers is not in the document]
Comment (timtmok) [reply src/node/desktop/src/core/winston-logger.ts:53]: Will do!

[the remark this answers is not in the document]
Comment (timtmok) [reply src/node/desktop/src/core/winston-logger.ts:53]: It's a central logging system on unix systems (usually viewable via /var/log/syslog). I think it's mostly useful for servers. For example, you can send logs to a logging server for better parsing and monitoring. OSX does have one as well but it isn't configurable (unless you jump through some security hoops). Although, the frontend isn't configured to use this, the rsession will pick up the setting and log to syslog.

File: src/node/desktop/src/core/winston-logger.ts [hunk 2]
```diff
@@ -76,15 +91,6 @@ export class WinstonLogger implements Logger {
   log(level: string, message: string): void {
     // log to default log locations
     this.logger.log(level, message);
-
-    // log to console in debug configurations
```
[the remark this answers is not in the document]
Comment (timtmok) [reply src/node/desktop/src/core/winston-logger.ts:80]: Good point. I'll add back similar logic to add another logging transport for stderr in a non-packaged build.

**Pull request merged**